transports/quic/tests/smoke.rs

```diff
@@ -182,6 +183,66 @@ fn concurrent_connections_and_streams_tokio() {
         .quickcheck(prop::<quic::tokio::Provider> as fn(_, _) -> _);
 }
 
+#[cfg(feature = "async-std")]
+#[async_std::test]
+async fn backpressure() {
+    let _ = env_logger::try_init();
+    let max_stream_data = quic::Config::new(&generate_tls_keypair()).max_stream_data;
+
+    let (mut stream_a, mut stream_b) = build_streams::<quic::async_std::Provider>().await;
+
+    let data = vec![0; max_stream_data as usize - 1];
+
+    stream_a.write_all(&data).await.unwrap();
+
+    let too_much_data = vec![0; 10];
+
+    assert!(
+        async_std::future::timeout(Duration::from_secs(1), stream_a.write_all(&too_much_data))
+            .await
+            .is_err()
+    );
+
+    let mut buf = vec![1; max_stream_data as usize - 1];
+    stream_b.read_exact(&mut buf).await.unwrap();
+
+    let mut buf = [0];
+    assert!(stream_b.read(&mut buf).now_or_never().is_none());
+}
+
+#[cfg(feature = "async-std")]
+#[async_std::test]
+async fn read_after_peer_dropped_stream() {
+    let _ = env_logger::try_init();
+    let (mut stream_a, mut stream_b) = build_streams::<quic::async_std::Provider>().await;
+
+    let data = vec![0; 10];
+
+    stream_b.close().now_or_never();
+    stream_a.write_all(&data).await.unwrap();
+    stream_a.close().await.unwrap();
+    drop(stream_a);
+
+    stream_b.close().await.unwrap();
+    let mut buf = Vec::new();
+    stream_b.read_to_end(&mut buf).await.unwrap();
+    assert_eq!(data, buf)
+}
+
+#[cfg(feature = "async-std")]
+#[async_std::test]
+#[should_panic]
+async fn write_after_peer_dropped_stream() {
+    let _ = env_logger::try_init();
+    let (stream_a, mut stream_b) = build_streams::<quic::async_std::Provider>().await;
+    drop(stream_a);
+    futures_timer::Delay::new(Duration::from_millis(1)).await;
+
+    let data = vec![0; 10];
+    stream_b.write_all(&data).await.expect("Write failed.");
+    stream_b.close().await.expect("Close failed.");
+}
```

Comment:
Can we move these to the muxer test harness? :)

Reply:
I am in favor of moving them to the test harness. 
However, contrary to the other tests in the harness these tests test a specific sequential order in which alice and bob do their calls, rather than the things running in parallel. So I think it would require some more "infrastructure" code and ideally we'd also add it for the other muxers. Since I don't want to block this PR on this I would propose to do it in a follow up PR.

Reply:
Right, that makes sense. I didn't consider that :)

Pushing the boundaries of the harness here 😁

Reply:
Tracked the idea in issue #3211.